src/Components/LifecycleChart/LifecycleChart.tsx

```diff
@@ -1,18 +1,75 @@
 import * as React from 'react';
 import '@patternfly/react-core/dist/styles/base.css';
 import { Chart, ChartAxis, ChartBar, ChartGroup, ChartTooltip, ChartVoronoiContainer } from '@patternfly/react-charts';
+import { AppLifecycleChanges } from '../../types/AppLifecycleChanges';
+import { SystemLifecycleChanges} from '../../types/SystemLifecycleChanges';
 
 interface LifecycleChartProps {
-  lifecycleData: { x: string; y0: Date; y: Date; packageType: string }[][];
+  lifecycleData: AppLifecycleChanges[] | SystemLifecycleChanges[];
+
 }
 
 const LifecycleChart: React.FC<LifecycleChartProps> = ({ lifecycleData }: LifecycleChartProps) => {
   const ref = React.useRef<HTMLDivElement>(null);
   const [chartWidth, setChartWidth] = React.useState(0);
 
+  //check data type and contruct a chart array 
+
+  const checkDataType = (lifecycleData: AppLifecycleChanges[] | SystemLifecycleChanges[]) => {
```

Comment:
We're going to have to adjust this anyway, so don't change this, but if we have a function like this in two places, we'd be better served with a utility function shared between the two components.